src/map/script.c

```diff
@@ -19501,6 +19501,12 @@ BUILDIN_FUNC(bg_get_data)
 	switch( type )
 	{
 		case 0: script_pushint(st, bg->count); break;
+		case 1:
+			for (i = 0; bg->members[i].sd != NULL; i++)
+				mapreg_setreg(reference_uid(add_str("$@arenamembers"), i), bg->members[i].sd->bl.id);
+			mapreg_setreg(reference_uid(add_str("$@arenamemberscount"), i), i);
```

Comment:
Like this it will create an array for arenamembercount and will put the actual count value at index i.
See getpartymember: `mapreg_setreg(add_str("$@partymembercount"),j);`